relay-server/src/services/buffer/envelope_buffer/mod.rs

```diff
@@ -131,6 +134,20 @@ impl PolymorphicEnvelopeBuffer {
             Self::InMemory(buffer) => buffer.has_capacity(),
         }
     }
+
+    /// Shuts down the [`PolymorphicEnvelopeBuffer`].
+    pub async fn shutdown(&mut self) -> bool {
+        // Currently, we want to flush the buffer only for disk, since the in memory implementation
+        // tries to not do anything and pop as many elements as possible within the shutdown
+        // timeout.
+        let Self::Sqlite(buffer) = self else {
```

Comment:
Decided to branch out here and not call any `flush` method on the memory impl, since I didn't want to make the flushing empty in the provider or stack, since it made no sense.

Reply:
Yes, that's a good point. I initially implemented it that way, but felt it was too verbose. But your argument makes it a more valid approach! Will fix